openpype/tools/workfiles/lib.py

```diff
@@ -0,0 +1,214 @@
+import os
+import shutil
+import uuid
+import time
+import json
+import logging
+import contextlib
+
+import appdirs
+
+
+class TempPublishFilesItem(object):
+    """Object representing on subfolder in app temp files.
+
+    Args:
+        item_id (str): Id of item used as subfolder.
+        data (dict): Metadata about temp files.
+        directory (str): Path to directory where files are copied to.
+    """
+
+    def __init__(self, item_id, data, directory):
+        self._id = item_id
+        self._directory = directory
+        self._filepath = os.path.join(directory, data["filename"])
+
+    @property
+    def directory(self):
+        return self._directory
+
+    @property
+    def filepath(self):
+        return self._filepath
+
+    @property
+    def id(self):
+        return self._id
+
+    @property
+    def size(self):
+        if os.path.exists(self.filepath):
+            s = os.stat(self.filepath)
+            return s.st_size
+        return 0
+
+
+class TempPublishFiles(object):
+    """Directory where """
+    minute_in_seconds = 60
+    hour_in_seconds = 60 * minute_in_seconds
+    day_in_seconds = 24 * hour_in_seconds
+
+    def __init__(self):
+        root_dir = appdirs.user_data_dir(
+            "published_workfiles_temp", "openpype"
+        )
+        if not os.path.exists(root_dir):
+            os.makedirs(root_dir)
```

Comment:
Why not use an actual temp directory?

Reply:
> Is that just so we could 'ease' "checking" a published workfile version without doing anything with it?

Yes

> why not directly allow "resaving" it to a new version

That'll probably be next PR. In some cases you don't want to save it to workfiles but just look into it.